src/en/news/blog/2025/rgw-improving-bucket-notifs/index.md

```diff
@@ -0,0 +1,218 @@
+
+Persistent Bucket Notifications were first rolled out in **Ceph Pacific** as part of the **RADOS Gateway (RGW) project**.  
+For a good background, see the [existing blog post on ceph.io](https://ceph.io/en/news/blog/2021/persistent-bucket-notifications-deep-dive/).  
+
+In brief, this post deals with **improving the performance** of persistent bucket notifications.  
+
+Currently, the implementation doesn’t fully leverage RADOS’ distributed capabilities. Each topic is tied to a **2 Phase Commit Queue** implemented as a   **single RADOS object**.
+
+This design creates a bottleneck. A **sharded queue implementation** allows notifications to be distributed across multiple RADOS objects, enabling **parallel writes via multiple OSDs**.  
+
+---
+
+<!-- ### Example  
+
+- Suppose you create **10,000 objects** in a Ceph bucket.  
+- In the current design, all `"Object Created"` notifications are directed to a **single queue object**, causing contention on a single OSD.  
+- With **sharded topic queues**, notifications are split across multiple queue objects. Multiple OSDs now handle writes in parallel, removing the bottleneck.  
+
+--- -->
+
+### What Was Done  
+
+- In the **old design**, each topic was mapped to **one RADOS object** (one `2pc_cls_queue`).  
+- In the **new design**, each topic maps to **multiple RADOS objects** (multiple `2pc_cls_queue` objects).  
+
+Each `2pc_cls_queue` object associated with a topic is called a **shard**.  
+
+The number of shards is configurable via:  
+
+```bash
+rgw_bucket_persistent_notif_num_shards
+```
+
+- **Default value:** 11  
+- **Note:** Existing topics are **not** re-sharded. To extend this feature for existing topics, you must delete and recreate the topic.  
+
+---
+
+### Topic Management  
+
+Topic operations updated to support sharded queues:  
+
+| Operation        | Behavior                                                                    |
+| ---------------- | --------------------------------------------------------------------------- |
+| **Create topic** | Creates multiple `2pc_cls_queue` objects as shards                          |
+| **Delete topic** | Deletes all associated shards                                               |
+| **Set topic**    | Supports toggling persistent ↔ non-persistent, with shard cleanup if needed |
+
+---
+
+### Design Details  
+
+#### Enqueue  
+- Ordering is guaranteed at the **per-key level** (per object in a bucket).  
+- The **target shard** is computed as:  
+
+```
+hash("bucket:object") % (# of shards)
+```  
+
+- Once computed, the notification is enqueued into the chosen shard (`2pc_cls_queue`).  
+  
+- The shards for a topic are named as per the follwing convention. 
+    - The first shard is just named as `topic_name`. This ensures that old RGW's unaware of shards can still enqueue notifications to a valid queue.
+    - The others Shards from 1 to (n - 1) are named as `topic_name.x`, where x is anywhere in `1` and `(n - 1)`.
```

Comment:
s/The others Shards/The other shards/

(Remove the s on the end of "others" and make "Shards" lowercase.)